portafly/src/utils/http.ts

```diff
@@ -1,35 +1,60 @@
+/* eslint-disable @typescript-eslint/no-throw-literal */
 import { getToken } from 'utils'
 
-interface HttpResponse<T> extends Response {
-  parsedBody?: T
+export type ValidationException = {
+  validationErrors: ValidationErrors
 }
 
+export type ValidationErrors = Record<string, string[]>
+
+// Anything else than a 200 should be considered an error.
 const fetchData = async <T>(request: Request): Promise<T> => {
-  const response: HttpResponse<T> = await fetch(request)
-  try {
-    response.parsedBody = await response.json()
-  } catch (ex) {
-    throw new Error(`The request ${request} didn't return a valid body`)
-  }
+  const response = await fetch(request)
 
   if (!response.ok) {
     throw new Error(response.statusText)
   }
-  return response.parsedBody as T
+
+  return await response.json() as T
+}
+
+// Watch for validation errors (422) and throw then in an error. Other than 201 is unexpected and
+// should be considered an error as well
+const postData = async <T>(request: Request, formData: FormData): Promise<T | any> => {
+  const body = Array.from(formData)
+    .reduce((acc, [key, value]) => acc.concat(`&${key}=${value}`), '')
+
+  const response = await fetch(request, { method: 'POST', body })
+
+  if (response.status === 422) {
+    const responseBody = await response.json()
+    throw { validationErrors: responseBody.errors } as ValidationException
+  }
+
+  if (response.status === 201) {
+    const created = await response.json() as T
+
+    return created
+  }
```

Comment:
This too, a more declarative way would be good when we need to extend it, or pass it around. Check my comment https://github.com/3scale/porta/pull/2037/files#r455059864

Reply:
This will be coped with in another PR